preflight_check/task.py

```diff
@@ -0,0 +1,484 @@
+                   b) Or, this node fenced.''', 'This node fenced'),
+        'pacemakerd': ('pacemakerd process restarted', None),
+    }
+
+    def  __init__(self, context):
+        self.target_kill = context.current_case
+        self.description = "Force kill {}".format(self.target_kill)
+        super(self.__class__, self).__init__(self.description, flush=True)
+        self.cmd = "killall -9 {}".format(self.target_kill)
+        self.looping = context.loop
+        if not self.looping:
+            self.expected = self.EXPECTED[self.target_kill][0]
+        else:
+            self.expected = self.EXPECTED[self.target_kill][1]
+        self.report = False
+        self.restart_happen_event = threading.Event()
+
+    def enable_report(self):
+        self.report = True
+        from . import main
+        if not os.path.isdir(main.ctx.report_path):
+            utils.msg_error("{} is not a directory".format(main.ctx.report_path))
+
+        report_path = main.ctx.report_path
+        report_name = "{}-{}.report".format(main.ctx.process_name, utils.now("%Y%m%d-%s"))
+        self.report_file = os.path.join(report_path, report_name)
+        print("(Report: {})".format(self.report_file))
+
+        if self.looping:
+            content_key = "{}-l".format(self.target_kill)
+        else:
+            content_key = self.target_kill
+
+        from . import explain
+        self.explain = explain.contents[content_key].format(nodeA=utils.this_node(), nodeB="other node")
+
+    def header(self):
+        h = '''==============================================
+Testcase:          {}
+Looping Kill:      {}
+Expected State:    {}
+'''.format(self.description, self.looping, self.expected)
+        return h
+
+    def to_json(self):
+        self.build_base_result()
+        self.result['Looping Kill'] = self.looping
+        self.result['Expected State'] = self.expected
+        from . import main
+        main.ctx.task_list = self.prev_task_list + [self.result]
+        utils.json_dumps()
+
+    def to_report(self):
+        if not self.report:
+            return
+        with open(self.report_file, 'w') as f:
+            f.write(self.header())
+            f.write("\nLog:\n")
+            for m in self.messages:
+                f.write("{} {}:{}\n".format(m[2], m[0].upper(), m[1]))
+            f.write("\nTestcase Explained:\n")
+            f.write("{}\n".format(self.explain))
+            f.flush()
+            os.fsync(f)
+
+    def pre_check(self):
+        """
+        Check the prerequisite
+        """
+        self.task_pre_check()
+        rc, pid = utils.get_process_status(self.target_kill)
+        if not rc:
+            raise TaskError("Process {} is not running!".format(self.target_kill))
+
+    def run(self):
+        while True:
+            rc, pid = utils.get_process_status(self.target_kill)
+            if rc:
+                self.info("Process {}({}) is running...".format(self.target_kill, pid))
```

Comment:
Is this necessary? Since have already run `pre_check` for the process status.

Reply:
Might be overlapped, but it has different concern, `pre_check` is to check the cluster is running, here is to check target process is running

Reply:
But anyway, it is the same `self.target_kill`